articles/Object-Detection-using-Fast-R-CNN-BrainScript.md

```diff
@@ -0,0 +1,537 @@
+
+## Summary
+
+<p align="center">
+<a target="_blank" href="./Tutorial_FastRCNN/svm_4WIN_20160803_12_37_07_Pro.jpg"><img src="./Tutorial_FastRCNN/svm_4WIN_20160803_12_37_07_Pro.jpg" alt="image" height="250"/></a>
+<a target="_blank" href="./Tutorial_FastRCNN/bus_01.jpg"><img src="./Tutorial_FastRCNN/bus_01.jpg" alt="image" height="250"/></a>
+</p>
+
+**This tutorial describes how to use CNTK Fast R-CNN with BrainScript and cntk.exe. Fast R-CNN using the CNTK Python API is described [here](./Object-Detection-using-Fast-R-CNN.md).**
+
+The above are examples images and object annotations for the grocery data set (left) and the Pascal VOC data set (right) used in this tutorial.
+
+_Fast R-CNN_ is an object detection algorithm proposed by _Ross Girshick_ in 2015.
+The paper is accepted to ICCV 2015, and archived at https://arxiv.org/abs/1504.08083.
+Fast R-CNN builds on previous work to efficiently classify object proposals using deep convolutional networks.
+Compared to previous work, Fast R-CNN employs a _region of interest pooling_ scheme that allows to reuse the computations from the convolutional layers.
+
+Additional material: a detailed tutorial for object detection using CNTK Fast R-CNN with BrainScript 
+(including optional SVM training and publishing the trained model as a Rest API) can be found [here](https://github.com/Azure/ObjectDetectionUsingCntk).
+
+## Setup
+
+To run the code in this example, you need a CNTK Python environment (see [here](./Setup-CNTK-on-your-machine.md) for setup help). You need to work from your Python 3.4 environment (if you are using Anaconda Python type `activate cntk-py34` (from a standard command line, not PowerShell), assuming `cntk-py34` is your environment name). Further you need to install a few additional packages. From your Python 3.4 environment (64bit version assumed, Python 3.5 analogously), go to the FastRCNN folder and run:
+
+```
+pip install -r requirements.txt
+```
+
+Known issue: to install scikit-learn you might have to run `conda install scikit-learn` if you use Anaconda Python.
+You will further need Scikit-Image and OpenCV to run these examples (and possibly numpy and scipy if your Python 3.4 package does not come with them).
+You need to download the corresponding wheel packages and install them manually. On Linux you can `conda install scikit-image opencv`.
+For Windows users, visit http://www.lfd.uci.edu/~gohlke/pythonlibs/, and download:
+
+* Python 3.4
+  * scikit_image-0.12.3-cp34-cp34m-win_amd64.whl
+  * opencv_python-3.1.0-cp34-cp34m-win_amd64.whl
+* Python 3.5
+  * scikit_image-0.12.3-cp35-cp35m-win_amd64.whl
+  * opencv_python-3.2.0-cp35-cp35m-win_amd64.whl
+
+Once you download the respective wheel binaries, install them with (Python 3.5 analogously):
+
+```
+pip install your_download_folder/scikit_image-0.12.3-cp34-cp34m-win_amd64.whl
+```
+
+Remark 1: if you see the message _No module named past_ when running the scripts please execute `pip install future`.
```

Comment:
>Remark 1:  [](start = 0, length = 10)

[!NOTE]

see for example: [https://docs.microsoft.com/en-us/cognitive-toolkit/Setup-Windows-Binary-Script](https://docs.microsoft.com/en-us/cognitive-toolkit/Setup-Windows-Binary-Script)